src/helpers/navigation.ts

```diff
@@ -0,0 +1,69 @@
+import { ContentFilter } from '../models/ContentFilter';
+import { Feed } from '../models/Feed';
+import { SubCategory, NewsSource } from '../models/recommendation/NewsSource';
+import { LocalFeed } from '../social/api';
+
+export interface Routes {
+    App: {};
+    Loading: {};
+    Welcome: {};
+    Post: {};
+    Root: {};
+    ProfileTab: {};
+    PostTab: {};
+    FavoriteTab: {};
+    SettingsTab: {};
+    AllFeedTab: {};
+    BugReportView: {};
+    SwarmSettingsContainer: {};
+    FilterListEditorContainer: {};
+    Backup: {};
+    Restore: {};
+    BackupRestore: {};
+    LogViewer: {};
+    Debug: {};
+    FeedListViewerContainer: {
+        showExplore: boolean,
+        feeds?: Feed[],
+    };
+    Feed: {
+        feedUrl: string,
+        name: string,
+    };
+    FeedInfo: {
+        feed: Feed;
+    };
+    EditFilter: {
+        filter: ContentFilter,
+    };
+    FeedSettings: {
+        feed: LocalFeed,
+    };
+    FeedFromList: {
+        feedUrl: string,
+        name: string,
+    };
+    CategoriesContainer: {
+
+    };
+    SubCategoriesContainer: {
+        title: string,
+        subCategories: SubCategory[],
+    };
+    NewsSourceFeed: {
+        feed: Feed,
+    };
+    NewsSourceGridContainer: {
+        newsSources: NewsSource[],
+        subCategoryName: string,
+    };
+    YourTab: {};
+}
+
+export interface TypedNavigation {
+    goBack: (routeKey?: string | null) => boolean;
```

Comment:
Here the type of routeKey should be `K` and K should be a generic parameter `<K extends keyof Routes>`.